framework/PostProcessors.py

```diff
@@ -1,5133 +0,0 @@
-
-#
-class ImportanceRank(BasePostProcessor):
-  """
-    ImportantRank class. It computes the important rank for given input parameters
-    1. The importance of input parameters can be ranked via their sensitivies (SI: sensitivity index)
-    2. The importance of input parameters can be ranked via their sensitivies and covariances (II: importance index)
-    3. The importance of input directions based principal component analysis of inputs covariances (PCA index)
-    3. CSI: Cumulative sensitive index (added in the future)
-    4. CII: Cumulative importance index (added in the future)
-  """
-  def __init__(self, messageHandler):
-    """
-      Constructor
-      @ In, messageHandler, message handler object
-      @ Out, None
-    """
-    BasePostProcessor.__init__(self, messageHandler)
-    self.targets = []
-    self.features = []
-    self.latent = []
-    self.latentDim = []
-    self.manifest = []
-    self.manifestDim = []
-    self.dimensions = []
-    self.mvnDistribution = None
-    self.acceptedMetric = ['sensitivityindex','importanceindex','pcaindex','transformation','inversetransformation','manifestsensitivity']
-    self.all = ['sensitivityindex','importanceindex','pcaindex']
-    self.statAcceptedMetric = ['pcaindex','transformation','inversetransformation']
-    self.what = self.acceptedMetric # what needs to be computed, default is all
-    self.printTag = 'POSTPROCESSOR IMPORTANTANCE RANK'
-    self.requiredAssObject = (True,(['Distributions'],[-1]))
-    self.transformation = False
-    self.latentSen = False
-    self.reconstructSen = False
-    self.pivotParameter = None # time-dependent pivot parameter
-    self.dynamic        = False # is it time-dependent?
-
-  def _localWhatDoINeed(self):
-    """
-      This method is local mirror of the general whatDoINeed method
-      It is implemented by this postprocessor that need to request special objects
-      @ In, None
-      @ Out, needDict, dict, list of objects needed
-    """
-    needDict = {'Distributions':[]}
-    needDict['Distributions'].append((None,self.mvnDistribution))
-    return needDict
-
-  def _localGenerateAssembler(self,initDict):
-    """
-      see generateAssembler method in Assembler
-      @ In, initDict, dict, dictionary ({'mainClassName':{'specializedObjectName':ObjectInstance}})
-      @ Out, None
-    """
-    distName = self.mvnDistribution
-    self.mvnDistribution = initDict['Distributions'][distName]
-
-  def _localReadMoreXML(self,xmlNode):
-    """
-      Function to read the portion of the xml input that belongs to this specialized class
-      and initialize some stuff based on the inputs
-      @ In, xmlNode, xml.etree.ElementTree Element Objects, the xml element node that will be checked against the available options specific to this Sampler
-      @ Out, None
-    """
-    paramInput = ImportanceRankInput()
-    paramInput.parseNode(xmlNode)
-    for child in paramInput.subparts:
-      if child.getName(
```

Comment:
These try blocks for `QTopologicalDecomposition` and `QDataMining` should have been added to the Factory. Without these, we cannot run the interactive test cases. @PaulTalbot-INL, I will add this fix to #175, since it is related to this whole refactoring theme.